Lib/test/test_array.py

```diff
@@ -40,6 +40,10 @@ def test_bad_constructor(self):
         self.assertRaises(TypeError, array.array, 'xx')
         self.assertRaises(ValueError, array.array, 'x')
 
+        # See bpo-43916
```

Comment:
Can you please write a sentence rather than just a reference to a bpo number? Same remark for all other comments.

Reply:
Will do

Reply:
Victor, I've rewritten this to use a more explicit approach, see [msg392040](https://bugs.python.org/issue43916#msg392040). I'm not sure it's worth it adding comments to the .c files now.

Reply:
With `{Py_tp_new, _PyType_DisabledNew}`, I think the code speaks for itself. Can I mark this as resolved?